src/Controllers/Common/ContentController.cs

```diff
@@ -780,7 +780,12 @@ public IActionResult GetAvatarByUserId([FromForm] Guid userId)
     [Produces("application/xml")]
     [Route("ContentWebService.asmx/GetUnselectedPetByTypes")] // used by old SoD (e.g. 1.13)
     [VikingSession(UseLock=false)]
-    public RaisedPetData[]? GetUnselectedPetByTypes(Viking viking, [FromForm] string petTypeIDs, [FromForm] bool active) {
+    public RaisedPetData[]? GetUnselectedPetByTypes(Viking viking, [FromForm] string? userId, [FromForm] string petTypeIDs, [FromForm] bool active, [FromForm] string apiKey) {
+        // Get viking based on userId, or use request player's viking as a fallback.
+        Guid? userIdGuid = userId!=null ? new Guid(userId) : null;
```

Comment:
maybe just `if (userId != null)` instead of null propagation

Reply:
Fixed